DFE-Digital/sts-plan-technology-for-your-school
# Feat: batch redis dependency key storage

## Overview

Follow on to [#234426](https://dfe-ssp.visualstudio.com/s190-schools-technology-services/_workitems/edit/234426)

Groups up the redis content dependencies and adds them in a single batch.
Anytime a CD changes something the page gets removed from the cache and has another first load, so we want the performance of this to be decent even if most of the time things will be coming out of the cache.

## Changes

### Minor
- Setting of dependencies is now done with one batched operation rather than many operations, which was making the first load of a page slow
- Also sets the timeout to 50ms instead of 2s to avoid waiting a long time when redis is down

## How to review the PR
Either
- Use your own instance of redis
- Or flush the data from the one in dev
- Try loading some pages for the first time in plan tech (with/without this PR) and see that page load times have improved

## Checklist
